Finish BrowseService and rename method to better names

topnessman opened the pull request:
Rename get/find to retrieve that better illustrates this process is
fetching existing staff from database

topnessman committed:
Finish BrowseService and rename method to better names

Rename get/find to retrieve that better illustrates this process is
fetching existing staff from database

dgcaccam reviewed (approved):
looks good

topnessman committed:
Add private modifier for some fields

dgcaccam commented on src/main/java/edu/uwaterloo/ece658/service/BrowseService.java:
```diff
@@ -0,0 +1,90 @@
+            // User explicitly subscribed to some tags => Directly show photos
+            // under these tags is ok. This includes UserTag that the user
+            // has subscribed to.
+            result.put(tag, tag.getPhotosUnderThisTag());
```
As an afterthought.... Where do we deal with private photos? As a private photo still may be under one of these tags. Will the front end not show these? Or should they be filtered out before passing the Map

topnessman replied:
Good question.
The point of private photo is only to say that "please don't notify my private photo to any one that subscribed to the tags of my private photo". notifyXXX() methods can expose private methods so we need to have a check there. Here this method can still leak the private photos. So I need to have a check, too.
Doing it at the backend should be correct. Even though you front-end doesn't show private photos, client can still find a way to get private photos if only your backend returns them.

topnessman committed:
Filter out private photos from result list in BrowseService

Rename isIsPrivate() to isPrivate()
User 1.8 version of source code since lambda expression is supported
since JDK1.8

dgcaccam reviewed (approved):
cool
